src/CommandLine.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_COMMANDLINE_H
+#define SQUID_COMMANDLINE_H
+
+#if HAVE_GETOPT_H
+#include <getopt.h>
+#endif
+#include <vector>
+
+typedef struct option LongOption;
+
+/// A struct option C++ wrapper, helps with option::name copying/freeing.
+class Option : public LongOption
```

Comment:
Having Option as a child of LongOption is rather confusing! To fix this, I suggest renaming LongOption to RawLongOption (if we need a name for that struct at all). I also recommend renaming Option to LongOption to distinguish it from typeless short options that we also store in the CommandLine objects.

Reply:
Done.